didact/camelk/first-integration.md

```diff
@@ -0,0 +1,109 @@
+
+### Step 1: Creating a Folder and Your Integration
+
+First, we need to set up our sample project. Camel K has many examples available at their [GitHub project hosting the source](https://github.com/apache/camel-k/tree/master/examples), but we're just going to use the simplest one: `simple.groovy`.
+
+You can download the file yourself from the GitHub repo and then copy it into a folder in your workspace. Or you can create a sample folder and copy in the file with the link below.
+
+- [ ] [Create a sample folder in your VS Code workspace and we will define it there.](didact://?commandId=vscode.didact.scaffoldProject&extFilePath=redhat.vscode-camelk/didact/camelk/simple-groovy-project.json&completion=Created%20simple-groovy%20project. "Creates a folder and copies in simple.groovy"){.didact}
+
+<details><summary>Advanced Users!</summary>
+
+If you simply want to get started writing some Groovy code, you can create a folder in your workspace, create a file called `simple.groovy`, and copy in the following code:
+
+```groovy
+// camel-k: language=groovy
+from('timer:groovy?period=1s')
+    .routeId('groovy')
+    .setBody()
+        .simple('Hello Camel K from ${routeId}')
+    .to('log:info?showAll=false')
+```
+
+</details>
+
+## Step 2: Exploring the Groovy integration file
+
+Now that you have an integration file, let's take a quick look at it. If you created the file yourself, go ahead and open it now. Go to the Explorer activity (Ctrl+Shift+E) and look at the workspace folders listed.
+
+If you clicked the link to create the sample project, you can find it in `first-camelk-integration/src/simple.groovy` in the Explorer view.
+
+- [ ] If you have the `first-camelk-integration` project in your workspace, you can [open the simple.groovy file in the editor.](didact://?commandId=vscode.openFolder&projectFilePath=first-camelk-integration/src/simple.groovy&completion=Opened%20the%20simple.groovy%20file "Opens the simple.groovy file"){.didact}
+
+For this file, we're simply telling Camel to put the message `Hello Camel K from ${routeId}` in the console once every second.
+
+## Step 3: Deploying the Integration
+
+![Camel K Start Integration menu](https://raw.githubusercontent.com/camel-tooling/vscode-camelk/master/images/camelk-start-integration-popup-menu.jpg){.imageRight}
+
+The `Tooling for Camel K` extension offers several tools to get your new integration started. 
+
+First, you can right-click the `simple.groovy` file and select `Start Apache Camel K Integration`. That will provide a drop-down in the Command palette area with a number of deployment options. In this case, select the `Dev Mode - Apache Camel K Integration in Dev Mode` option. 
+
+- [ ] [Start the simple.groovy integration in Dev Mode](didact://?commandId=camelk.startintegration&projectFilePath=first-camelk-integration/src/simple.groovy&text=Dev%20Mode&completion=Deployed%20the%20simple.groovy%20file%20in%20Dev%20mode. "Deploys the simple.groovy file in 'Dev mode'"){.didact}
```

Comment:
this is not working if the file wasn't created in first-camelk-integration folder. it is failing silently in this case.
Would be nice to provide at least a warning or explain in text that it needs to follow the pattern (the best would be to detect the simple.groovy file in the workspace but looks alike a new feature and can be done in another iteration)

Reply:
+1

VSCode has showed already obscure implicit assumptions related to folders a project roots. 

Ideally we want to push warnings up to Didact itself or reason on macros to be used by the document author to help in recurring situations like these.